### DISCUSSION TITLE
sympy.physics.mechanics: Warning coming sometimes

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#26792] [Q&A]
I am trying a simulation.
When setting up Kane's equations of motion, I get this error:

UserWarning: Loops are defined among the orientation of frames. This is likely not desired and may cause errors in your calculations.

But the I run the cell again, and there is no error. It seems to come erratically, at least I found no pattern.

**Thanks for any explanation!**

**Comment by moorepants** [comment]
If you are using Jupyter and you "restart and run all" you should not see this warning. If you are working interactively and redefining things, this warning may be displayed.

**Reply by moorepants** [reply]
The warning appears when you do something like:

```python
A.orient_axis(N)
B.orient_axis(A)
B.orient_axis(N)
```

**Reply by Peter230655** [reply]
> If you are using Jupyter and you "restart and run all" you should not see this warning. If you are working interactively and redefining things, this warning may be displayed.

Exactly right! But when I run the cell twice, it also seems to come once, but not again...

**Reply by Peter230655** [reply]
> The warning appears when you do something like:
> 
> ```python
> A.orient_axis(N)
> B.orient_axis(A)
> B.orient_axis(N)
> ```

I checked, but I do not think, I did anything this stupid. Actually I do this:

```
udisc = Omega * (1 - sm.exp(-alpha * t))
qdisc = Omega * t + Omega * sm.exp(-alpha * t) / alpha - Omega / alpha
A2.orient_axis(N, qdisc, N.z)
A2.set_ang_vel(N, udisc*N.z)

A1.orient_body_fixed(A2, (q1, q2, q3), '123')
rot = A1.ang_vel_in(N)
A1.set_ang_vel(A2, u1*A1.x + u2*A1.y + u2*A1.z)
rot1 = A1.ang_vel_in(N)
```
(I one of your examples of your lectureI saw a video of a ball rolling on a horizontal spinning disc. I am trying to simulate this. Actually I want to use opty to get the ball to the center of the disk)

**Reply by Peter230655** [reply]
> If you are using Jupyter and you "restart and run all" you should not see this warning. If you are working interactively and redefining things, this warning may be displayed.

This is how it seems to happen. 
Anyway, it seems to build the EOMs
Should I close it?
**Thanks!**

**Reply by moorepants** [reply]
If you "restart and run all" and the warning is displayed, then it means you are likely defining your system incorrectly. That's why we display the warning.

**Comment by Peter230655** [comment]
I does **not** come when I "restart and run all".
Only _sometimes_, when I run the cell where Kane's EOMs are formed. 
I can easily attach the part of the program, if you want to see it.

**Reply by moorepants** [reply]
Then it is caused by you creating undesired loops.

**Reply by Peter230655** [reply]
> Then it is caused by you creating undesired loops.

But should this undesired loop not be present also when I use "restart and run all"?
**Thanks!**

**Reply by moorepants** [reply]
I wrote:

> If you are using Jupyter and you "restart and run all" you should not see this warning.

> If you "restart and run all" and the warning is displayed, then it means you are likely defining your system incorrectly. That's why we display the warning.

I don't know how to say it much differently.

**Comment by Peter230655** [comment]
Now totally clear! I did not connect properly.
**Thanks!**
